src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -18,6 +19,9 @@ public abstract partial class ComWrappers
         internal static Guid IID_IUnknown = new Guid(0x00000000, 0x0000, 0x0000, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
 
         private readonly ConditionalWeakTable<object, ManagedObjectWrapperHolder> _ccwTable = new ConditionalWeakTable<object, ManagedObjectWrapperHolder>();
+        private readonly Lock _lock = new Lock();
+        private readonly Dictionary<IntPtr, WeakReference> _rcwCache = new Dictionary<IntPtr, WeakReference>();
```

Comment:
Do you also need code to delete entries from this cache?

Also, the value can be GCHandle. It will save you an extra allocation.

Reply:
Probably I should not remove entries from the cache, if I want to maintain compatibility with .NET Core. I think idea for that is to maintain identity between Com Object and .NET Object.

Reply:
I am pretty sure .NET Core removes entries from the cache once things become unreachable.

Otherwise, this cache would be a guaranteed memory leak.

Reply:
Am I read this incorrectly? https://github.com/dotnet/runtime/blob/main/src/tests/Interop/COM/ComWrappers/API/Program.cs#L118

Reply:
This is for the other direction (CCWs). It is not really always present. It is only present for as long as the object instance is alive.

Reply:
Okay. Make sense.